test/e2e/mnist_rayjob_raycluster_test.go

```diff
@@ -388,10 +393,37 @@ func constructRayCluster(_ Test, namespace *corev1.Namespace, mnist *corev1.Conf
 			},
 		},
 	}
+	if accelerator.IsGpu() {
+		// Add toleration for the GPU
+		raycluster.Spec.WorkerGroupSpecs[0].Template.Spec.Tolerations = append(raycluster.Spec.WorkerGroupSpecs[0].Template.Spec.Tolerations, corev1.Toleration{
+			Key:      accelerator.ResourceLabel,
+			Operator: corev1.TolerationOpExists,
+		})
+		// Add GPU resource quota
+		raycluster.Spec.WorkerGroupSpecs[0].Template.Spec.Containers[0].Resources.Requests[corev1.ResourceName(accelerator.ResourceLabel)] = resource.MustParse("1")
+		raycluster.Spec.WorkerGroupSpecs[0].Template.Spec.Containers[0].Resources.Limits[corev1.ResourceName(accelerator.ResourceLabel)] = resource.MustParse("1")
+	}
+
+	return &raycluster
 }
 
-func constructRayJob(_ Test, namespace *corev1.Namespace, rayCluster *rayv1.RayCluster, accelerator string, numberOfGpus int) *rayv1.RayJob {
-	return &rayv1.RayJob{
+func constructRayJob(_ Test, namespace *corev1.Namespace, rayCluster *rayv1.RayCluster, accelerator Accelerator, rayImage string) *rayv1.RayJob {
+	pipPackages := []string{
+		"pytorch_lightning==2.4.0",
+		"torchmetrics==1.6.0",
+		"torchvision==0.19.1",
+	}
+
+	// Append AMD-specific packages
+	if accelerator.ResourceLabel == "amd.com/gpu" {
```

Comment:
You can directly use here `if accelerator == AMD {`

Reply:
yes right, updated it